### PR TITLE
add test for the version of openrosa-xpath-evaluator

### PR DESCRIPTION
**Pull request by njogz**
# Description

Add a test which runs against the version of `openrosa-xpath-evaluator`

medic/cht-core#6490

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Review by garethbowen — CHANGES_REQUESTED** [review]
Good start. This will definitely catch if we bump the version.

I was thinking the test would actually check using the openrosa-xpath-evaluator to concatenate some strings, so it's testing the feature rather than the version. We don't know when this will break so it might be that there are a couple of minors which would fail this test but still allow string concatenation so it'd give a false negative.

I think you could find a way to test the [OpenrosaXpathEvaluatorBinding](https://github.com/medic/cht-core/blob/master/webapp/src/js/enketo/OpenrosaXpathEvaluatorBinding.js) module directly which is called by Enketo.

Also it'd be worth having a comment as to _why_ this test will start failing, something like:

```
// If this test starts failing then we need to document in the release notes that we've removed the deprecated feature allowing for concatenation of strings
```

File: tests/e2e/concatenate-xpath-strings.spec.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    // select form
+    const addButton = element(
+      by.css('.action-container .general-actions:not(.ng-hide) .fa-plus')
+    );
+    helper.waitElementToPresent(addButton);
+    helper.clickElement(addButton);
+    element(
+      by.css(
+        '.action-container .general-actions .dropup.open .dropdown-menu li:first-child a'
```
Comment (garethbowen) [inline tests/e2e/concatenate-xpath-strings.spec.js:48]: This seems flaky. If another form is configured and happens to be first it'll start failing. Is there another selector you can use to find the exact form you want to fill in?

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice one!

I've got one suggestion to hopefully make the test more reliable but then it'll be good to go!

File: tests/e2e/concatenate-xpath-strings.spec.js (hunk 1 above)
Comment (njogz) [reply tests/e2e/concatenate-xpath-strings.spec.js:48]: I haven't been able to find a more specific selector yet. This same selector is in multiple tests e.g. https://github.com/medic/cht-core/blob/master/tests/page-objects/forms/generic-form.po.js#L59 and https://github.com/medic/cht-core/blob/master/tests/e2e/forms/submit-photo-upload-form.spec.js#L47. Maybe we should add a new issue to update all such tests?

**Review by garethbowen — APPROVED** [review]
Nice one.

File: tests/e2e/concatenate-xpath-strings.spec.js (hunk 1 above)
Comment (garethbowen) [reply tests/e2e/concatenate-xpath-strings.spec.js:48]: Ok to can leave it as is for now.

**Pull request merged**